### ISSUE TITLE
does this work on the server side?

### ISSUE THREAD

**Issue by geekyme** [webpack/items-store#3]
specifically isomorphic web apps? the https://github.com/spoike/refluxjs project is really user friendly but wat it lacks is server side compatibility

**Comment by sokra** [comment]
yes, I've used reflux before, but found it difficult/impossible to implement server-side prerendering. This is designed to support it.

Here is an example: https://github.com/webpack/react-starter

The basic idea is:

* You have stores for the client, which fetch data i. e. with XHR
* and you have the same stores for the server, which fetch data i. e. from a DB.
* The server stores at created per request
* after rendering you grab the cached data for the stores and embed it into the HTML page
* Client stores are prefilled with that data.

https://github.com/webpack/react-starter/blob/master/config/prerender.jsx
https://github.com/webpack/react-starter/blob/master/lib/server.js#L119-L131
https://github.com/webpack/react-starter/blob/master/app/mainStores.js#L65
https://github.com/webpack/react-starter/blob/master/app/prerender.html#L8-L11

**Comment by geekyme** [comment]
I see. Interesting. I’ll take a deep dive soon. 
Any chance of a state leakage between concurrent requests in the stores?

**Comment by sokra** [comment]
An ItemStore just fetch data from a higher instance. You have separate stores per request (because the data is serialized to the client too). So any communication between concurrent requests is managed by the "higher instance".

**Comment by jonaswindey** [comment]
This is great work. I was looking at Fluxible.io, but I like your approach a lot more and with less overhead.  Keep it up!
